src/tasty/tasty.tsx

```diff
@@ -184,27 +188,32 @@ function tasty<K extends StyleList, C = Record<string, unknown>>(
       propStyles = null;
     }
 
+    if (!styles || (styles && Object.keys(styles).length === 0)) {
+      styles = undefined;
+    }
+
     const useDefaultStyles = !propStyles && !styles;
+    const styleCacheKey = `${styles ? JSON.stringify(styles) : ''}${
```

Comment:
It'll run stringify on each render, are we sure it's less expensive that merging styles?

Reply:
Surprisingly there are no performance issues with that but I've just improved that part to avoid stringification.